errors.go

```diff
@@ -2,147 +2,25 @@ package fuego
-	title := e.Title
-	if title == "" {
-		title = http.StatusText(code)
-		if title == "" {
-			title = "HTTP Error"
-		}
-	}
-	msg := fmt.Sprintf("%d %s", code, title)
-
-	detail := e.DetailMsg()
-	if detail == "" {
-		return msg
-	}
-
-	return fmt.Sprintf("%s: %s", msg, e.Detail)
-}
-
-func (e HTTPError) StatusCode() int {
-	if e.Status == 0 {
-		return http.StatusInternalServerError
-	}
-	return e.Status
-}
-
-func (e HTTPError) DetailMsg() string {
-	return e.Detail
-}
-
-func (e HTTPError) Unwrap() error { return e.Err }
-
-// BadRequestError is an error used to return a 400 status code.
-type BadRequestError HTTPError
-
-var _ ErrorWithStatus = BadRequestError{}
-
-func (e BadRequestError) Error() string { return e.Err.Error() }
-
-func (e BadRequestError) StatusCode() int { return http.StatusBadRequest }
-
-func (e BadRequestError) Unwrap() error { return HTTPError(e) }
 
-// NotFoundError is an error used to return a 404 status code.
-type NotFoundError HTTPError
-
-var _ ErrorWithStatus = NotFoundError{}
-
-func (e NotFoundError) Error() string { return e.Err.Error() }
-
-func (e NotFoundError) StatusCode() int { return http.StatusNotFound }
-
-func (e NotFoundError) Unwrap() error { return HTTPError(e) }
-
-// UnauthorizedError is an error used to return a 401 status code.
-type UnauthorizedError HTTPError
-
-var _ ErrorWithStatus = UnauthorizedError{}
-
-func (e UnauthorizedError) Error() string { return e.Err.Error() }
-
-func (e UnauthorizedError) StatusCode() int { return http.StatusUnauthorized }
-
-func (e UnauthorizedError) Unwrap() error { return HTTPError(e) }
-
-// ForbiddenError is an error used to return a 403 status code.
-type ForbiddenError HTTPError
-
-var _ ErrorWithStatus = ForbiddenError{}
-
-func (e ForbiddenError) Error() string { return e.Err.Error() }
-
-func (e ForbiddenError) StatusCode() int { return http.StatusForbidden }
-
-func (e ForbiddenError) Unwrap() error { return HTTPError(e) }
-
-// ConflictError is an error used to return a 409 status code.
-type ConflictError HTTPError
-
-var _ ErrorWithStatus = ConflictError{}
-
-func (e ConflictError) Error() string { return e.Err.Error() }
-
-func (e ConflictError) StatusCode() int { return http.StatusConflict }
-
-func (e ConflictError) Unwrap() error { return HTTPError(e) }
-
-// InternalServerError is an error used to return a 500 status code.
-type InternalServerError = HTTPError
-
-// NotAcceptableError is an error used to return a 406 status code.
-type NotAcceptableError HTTPError
-
-var _ ErrorWithStatus = NotAcceptableError{}
-
-func (e NotAcceptableError) Error() string { return e.Err.Error() }
-
-func (e NotAcceptableError) StatusCode() int { return http.StatusNotAcceptable }
+	"github.com/go-fuego/fuego/internal"
+)
 
-func (e NotAcceptableError) Unwrap() error { return HTTPError(e) }
+type (
+	ErrorWithStatus     = internal.ErrorWithStatus
+	ErrorWithDetail     = internal.ErrorWithDetail
+	HTTPError           = internal.HTTPError
```

Comment:
I'm unsure what will be provided in the public documentation.

It needs to be double checked.

Unless you want to deprecate them all, but then all of them need to be clearer.

Reply:
I think we should provide (by copying) a documentation at this level too